### PR TITLE
mobile: allow building out stats reporting

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low - should be a no op for the default build
Testing: adapted a few tests, 
Docs Changes: n/a
Release Notes: inline

### CODE REVIEW

File: mobile/envoy_build_config/BUILD [hunk 1]
```diff
@@ -56,6 +52,14 @@ envoy_cc_library(
             "@envoy//source/common/quic:quic_transport_socket_factory_lib",
         ],
         "@envoy",
+    ) + envoy_select_envoy_mobile_stats_reporting(
+        [
+            "@envoy//source/extensions/stat_sinks/metrics_service:config",
+            "@envoy//source/extensions/stat_sinks/statsd:config",
+            "@envoy//source/extensions/clusters/logical_dns:logical_dns_cluster_lib",
```
Comment (RyanTheOptimist) [inline mobile/envoy_build_config/BUILD:59]: Don't we need this cluster for happy eyeballs?

File: mobile/envoy_build_config/BUILD [hunk 2]
```diff
@@ -56,6 +52,14 @@ envoy_cc_library(
...
+            "@envoy//source/extensions/clusters/static:static_cluster_lib",
```
Comment (RyanTheOptimist) [inline mobile/envoy_build_config/BUILD:60]: This doesn't seem obviously related to stats, but I'm guessing this is used to implement the stats cluster?

**Review by RyanTheOptimist — COMMENTED** [review]
Thanks for doing this! Perfect timing!

**Comment by Augustyniak** [comment]
>low - should be a no op for the default build

by no op you mean that by default stats support will be compiled out of the binary, right?

File: mobile/envoy_build_config/BUILD (hunk 1 above)
Comment (alyssawilk) [reply mobile/envoy_build_config/BUILD:59]: Happy eyeballs works with DFP, no?

Looks to me like we use logical DNS for direct responses (in swift tests) and for stats clusters (library/cc/engine_builder.cc LOGICAL_DNS)

File: mobile/envoy_build_config/BUILD (hunk 2 above)
Comment (alyssawilk) [reply mobile/envoy_build_config/BUILD:60]: actually looks like this one can go in my TODO test only extension PR

**Review by alyssawilk — COMMENTED** [review]
correct - stats should be on for the default build, and is "verified" by CI because whatever kotlin test verifies yaml for the stats cluster fails with stats disabled - the kotlin APIS are not compiled out and the no-op function then fails string matching.

File: mobile/envoy_build_config/BUILD (hunk 1 above)
Comment (Augustyniak) [reply mobile/envoy_build_config/BUILD:59]: another reason to cut direct responses from our APIs. Hopefully we will have time to prioritize this on our end (Lyft) sooner than later.

**Review by Augustyniak — DISMISSED** [review]

**Comment by Augustyniak** [comment]
we may add to add a mention of this to the changelog

File: mobile/envoy_build_config/BUILD (hunk 1 above)
Comment (RyanTheOptimist) [reply mobile/envoy_build_config/BUILD:59]: I'm sure you're right, but just to make sure I understand... The docs talk about LOGICAL_DNS, but I guess that's the DiscoveryType not the cluster type?
https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/connection_pooling#happy-eyeballs-support
And in one of the PRs which wires up Happy Eyeballs, we modifed logical_host.cc, but that's also not the same thing as the logical DNS *cluster*?
https://github.com/envoyproxy/envoy/pull/17371/files#diff-83c925d22248fe7ab6cad6424953960a158f879f8a1bfa45b510c64ccceb913b

So nothing above requires the logical_dns_cluster to work, right?

**Review by RyanTheOptimist — DISMISSED** [review]
LGTM, assuming I understand the logical DNS cluster stuff.

(And +1 to rafal's suggestion of adding this to the changelog)

File: mobile/envoy_build_config/BUILD (hunk 1 above)
Comment (alyssawilk) [reply mobile/envoy_build_config/BUILD:60]: yeah for DFP we configure the same `ALL` but in the cache config which goes in the DFP dns_cache_config, boostrap config, etc.

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**